### DISCUSSION TITLE
[OP Stack Deployment] L1 RPC rate limiting causing L2 reorgs

### DISCUSSION THREAD

**Discussion by bap2pecs** [ethereum-optimism/developers#671] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

we have created our own OP devnet but meet some L2 reorgs. We also see our L1 RPC got rate limited. So we suspect it's related.

Can anyone help confirm that:

1) if the L1 rpc gets rate limited, can it lead to L2 reorgs?
2) we noticed our L2 devnet batcher freaked out for a while and kept submitting batch txs every L1 block. could it be also due to L1 rpc rate limiting?

**Comment by hash1go** [comment]
Let me check about the context of your question - are you using a third-party L1 RPC provider?

**Reply by bap2pecs** [reply]
> Let me check about the context of your question - are you using a third-party L1 RPC provider?

yes

**Reply by hash1go** [reply]
It's difficult to answer about OP Stack's behavior when L1 RPC suddenly becomes unavailable.

Also, due to cost and rate limit issues, I recommend that L2 chain operators run their own L1 nodes. With a server that has sufficient specifications, L1 and L2 should be able to coexist on the same host.

**Reply by bap2pecs** [reply]
> It's difficult to answer about OP Stack's behavior when L1 RPC suddenly becomes unavailable.
> 
> 
> 
> Also, due to cost and rate limit issues, I recommend that L2 chain operators run their own L1 nodes. With a server that has sufficient specifications, L1 and L2 should be able to coexist on the same host.

got it thanks! we are running our own node now.

tho i am still curious on the exact behaviour

**Reply by hash1go** [reply]
Could you share your logs and configurations?

**Reply by bap2pecs** [reply]
> Could you share your logs and configurations?

sure but which particular parts?

**Reply by hash1go** [reply]
I'd like to check the configs and error logs of the main components such as op-geth and op-node.

**Comment by Chomtana** [comment, marked as answer]
Note: This answer is my opinion, it may not correct.

> if the L1 rpc gets rate limited, can it lead to L2 reorgs?

If rate limiting prevent batcher or proposer to submit to the L1 for a very long time, it has a chance that L2 will perform deep reorgs that's really hard to recover.

But if it's being rate limited but batcher and proposer can still submit to the L1, it's unlikely that rate limit will cause L2 to reorg.

> we noticed our L2 devnet batcher freaked out for a while and kept submitting batch txs every L1 block. could it be also due to L1 rpc rate limiting?

Yes, if batcher get rate limited while submitting batch tx, it may increase gas price and keep retrying.

**Reply by bap2pecs** [reply]
ok we just got our L1 node synced and restarted the batcher. the batcher initially sent some txs but then stopped:

```
t=2024-12-12T01:28:20+0000 lvl=info msg="Publishing transaction" service=batcher tx=0x024d14605ef753c654dde625f07c5e6910ac1e561878c4518c8f12f3057f4fbe nonce=12947 gasTipCap=4000000000 gasFeeCap=28657309960 gasLimit=21000 blobs=1 blobFeeCap=268747933588 tx=0x024d14605ef753c654dde625f07c5e6910ac1e561878c4518c8f2f3057f4fbe
t=2024-12-12T01:28:20+0000 lvl=info msg="bumping gas price for transaction" service=batcher tx=0x024d14605ef753c654dde625f07c5e6910ac1e561878c4518c8f12f3057f4fbe nonce=12947 gasTipCap=4000000000 gasFeeCap=28657309960 gasLimit=21000 blobs=1 blobFeeCap=268747933588
t=2024-12-12T01:28:21+0000 lvl=error msg="unable to increase gas" service=batcher tx=0x024d14605ef753c654dde625f07c5e6910ac1e561878c4518c8f12f3057f4fbe nonce=12947 gasTipCap=4000000000 gasFeeCap=28657309960 gasLimit=21000 blobs=1 blobFeeCap=268747933588 err="bumped blob fee 537495867176 is over 5x multiple of the suggested value: blob fee limit reached"
````

any idea how to recover this?

**Reply by opfocus** [reply]
If you still encounter this problem, please open a separate discussion and this discussion will be closed.
